proposals/0257-eliminate-vote-fees.md

```diff
@@ -0,0 +1,362 @@
+rewards that they are eligble for. To determine this we first define the maximum
+stake weighted vote credits $M$ that they can earn in an epoch where they produce
+$L$ leader slots. Here $s_i$ is the validator's stake, and $S$ is the total stake.
+
+$$M = 12 s_i \times 432000 + 4 L S$$
+
+The first term represents the maximum voting credits $3/4$ of the TVC credits
+(16), and the second term is the maximum leader inclusion credits covered in the
+following section.
+
+Their rewards in each epoch will be based on the fraction of the stake weighted
+credits $c_{i}$ they actually earn relative to this maximum $M$ multiplied by the
+total inflation rewards available in the epoch $I$.
+
+$$ R_i = I \frac{c_i}{S M} $$
+
+Any unearned sol rewards will be burned.
+
+This represents a very slight reduction in total rewards available to validators:
+
+![rewards table](../assets/0257-rewardstbl.png)
+
+### Vote Inclusion Credits
+
+25% of issuance will be redirected from vote credits to vote inclusion credits.
+These credits will be awarded to validators based on the number of votes they
+include in a timely manner. Each vote inclusion credit is awarded to the leader
+and is worth 1/3 of the vote credits awarded to the validator that the vote belongs
+to.
+
+For example assume that a latency 1 vote comes in from voter $v$ to be included
+in leader $l$'s block. $v$ will earn 12 credits based on their stake weight, and
+$l$ will earn 4 credits based on $v$'s stake weight:
+
+$c_v := c_v + 12 s_v$
+
+$c_l := c_l + 4 s_v$
+
+Note that this accounting matches the $M$ derived above, if $v$ votes perfectly
+and includes everyones votes (including their own):
+
+$$c_v = 12 s_v \times 432000 + 4 L \sum_{i\in V} s_i = M$$
+
+In order to implement this change, the vote program will track credits $c_v$ as
+stake weighted totals.
+
+### Vote transaction CU accounting
+
+Vote transactions will no longer count towards the global CU limit in the block,
+they will have their own limit of one vote per validator per block. With SIMD-0218
+IVC, there is no longer any benefit to including multiple old votes within a single
+block, instead only the latest vote will be used.
+
+### Validator set size
+
+By removing the fees for votes, we greatly reduce the barrier for entry as a voter.
+This allows attackers to spin up numerous minimally staked nodes and spam votes
+in an effort to slow down or DoS the cluster.
+
+In order to address this we choose to limit the number of validators that can vote
+at any given time.
+
+These validators that are chosen to vote are the top 2000 validators by stake,
+rotating every epoch.
+
+We must abide by the maximum of 18% of the total cluster stake movement allowed
+per epoch,
+Priority will be given to changes in stake on validators that are already within
+the top 2000 validators by stake before adjustments from inactive validators
+becoming active or active validators becoming inactive.
```

Comment:
Is it always the case that the heaviest staked 2000 validators get in the next set?  I assume that's the case but I want to be sure.  It might be useful to add more information on how this affects the leader schedule calculation since those are pre-computed. 

Also, I assume that this does create a minimum stake amount to participate in consensus, but it's dynamic and market driven rather than a set amount.

Reply:
It would be the heaviest, but a rotation into the validator set counts as part of the 9%/9% limit. So if say in an extreme scenario 20% of stake was activating on the 2001st validator and there's already 6% stake activating across the other validators, in order to maintain the cap it might take multiple epochs for this 2001st validator's stake to activate and for it to be recognized as the 2000th validator